nilearn/glm/_base.py

```diff
@@ -529,26 +529,28 @@ def generate_report(
             )
             first_level_contrast = None
 
-        if not hasattr(self, "_reporting_data"):
-            self._reporting_data: dict[str, Any] = {
-                "trial_types": [],
-                "noise_model": getattr(self, "noise_model", None),
-                "hrf_model": getattr(self, "hrf_model", None),
-                "drift_model": None,
-            }
         warning_messages = []
 
         sig = inspect.signature(self.generate_report).parameters
         parameters = dict(**sig)
-        if height_control is not None and float(threshold) != float(
-            parameters["threshold"].default
+        # if user specifies threshold
+        if (
+            float(threshold) != float(parameters["threshold"].default)
+            and height_control is not None
         ):
             warning_messages.append(
-                f"\n'{threshold=}' is not used with '{height_control=}'."
+                f"'{threshold=}' is not used with '{height_control=}'."
                 "\n'threshold' is only used when 'height_control=None'. "
-                f"\nSet 'threshold' to '{parameters['threshold'].default}' "
-                "to avoid this warning."
+                "\nSetting 'height_control' to None. "
             )
+            height_control = None
+        elif height_control is not None:
+            warning_messages.append(
+                f"'{threshold=}' is not used with '{height_control=}'."
+                "\n'threshold' is only used when 'height_control=None'. "
+                "\nSetting 'threshold' to None. "
+            )
+            threshold = None
```

Comment:
If `threshold` value is not changed, I assume that `height_control` has priority over threshold, and set `threshold=None`. Is this reasonable? 
I think if we fix the default values, we won't have this problem. We will warn the user, only depending on the parameters specified.